repo-scripts/size-analysis/analysis-helper.ts

```diff
@@ -0,0 +1,316 @@
+              replaceAll(
+                reExportsMember,
+                reExportedSymbol,
+                exportSpecifier.name.escapedText.toString()
+              );
+            } else {
+              actualExports.push(reExportedSymbol);
+            }
+          });
+          filterAllBy(reExportsMember, actualExports);
+        }
+        // concatenate re-exported MemberList with MemberList of the dts file
+        declarations.functions.push(...reExportsMember.functions);
+        declarations.variables.push(...reExportsMember.variables);
+        declarations.classes.push(...reExportsMember.classes);
+        declarations.enums.push(...reExportsMember.enums);
+      }
+    }
+  });
+
+  if (map) {
+    declarations = mapSymbolToType(map, declarations);
+    declarations = dedup(declarations);
+  }
+
+  // Sort to ensure stable output
+  declarations.functions.sort();
+  declarations.classes.sort();
+  declarations.variables.sort();
+  declarations.enums.sort();
+
+  return declarations;
+}
+/**
+ * To Make sure symbols of every category are unique.
+ */
+function dedup(memberList: MemberList): MemberList {
+  const classesSet: Set<string> = new Set(memberList.classes);
+  memberList.classes = Array.from(classesSet);
+
+  const functionsSet: Set<string> = new Set(memberList.functions);
+  memberList.functions = Array.from(functionsSet);
+
+  const enumsSet: Set<string> = new Set(memberList.enums);
+  memberList.enums = Array.from(enumsSet);
+
+  const variablesSet: Set<string> = new Set(memberList.variables);
+  memberList.variables = Array.from(variablesSet);
+
+  return memberList;
+}
+
+function mapSymbolToType(
+  map: Map<string, string>,
+  memberList: MemberList
+): MemberList {
+  const newMemberList: MemberList = {
+    functions: [],
+    classes: [],
+    variables: [],
+    enums: []
+  };
+  memberList.classes.forEach(element => {
+    if (map.has(element)) {
+      newMemberList[map.get(element)].push(element);
+    } else {
+      newMemberList.classes.push(element);
+    }
+  });
+  memberList.variables.forEach(element => {
+    if (map.has(element)) {
+      newMemberList[map.get(element)].push(element);
+    } else {
+      newMemberList.variables.push(element);
+    }
+  });
+  memberList.enums.forEach(element => {
+    if (map.has(element)) {
+      newMemberList[map.get(element)].push(element);
+    } else {
+      newMemberList.enums.push(element);
+    }
+  });
+  memberList.functions.forEach(element => {
+    if (map.has(element)) {
+      newMemberList[map.get(element)].push(element);
+    } else {
+      newMemberList.functions.push(element);
+    }
+  });
+  return newMemberList;
+}
+
+function isReExported(symbol: string, reExportedSymbols: string[]): boolean {
+  return reExportedSymbols.includes(symbol);
+}
+
+function extractRealSymbolName(exportSpecifier: ts.ExportSpecifier): string {
+  // if property name is not null -> export is renamed
+  if (exportSpecifier.propertyName) {
```

Comment:
is the propertName the renamed name, or the original name? can use a comment here.